packages/cursorless-engine/src/actions/Wrap.ts

```diff
@@ -50,67 +45,45 @@ export default class Wrap {
           },
         ]);
 
-        const delimiterSelectionInfos: FullSelectionInfo[] = boundaries.flatMap(
-          ({ start, end }) => {
-            return [
-              getSelectionInfo(
-                document,
-                start,
-                RangeExpansionBehavior.openClosed,
-              ),
-              getSelectionInfo(
-                document,
-                end,
-                RangeExpansionBehavior.closedOpen,
-              ),
-            ];
-          },
-        );
-
-        const cursorSelectionInfos = editor.selections.map((selection) =>
-          getSelectionInfo(
-            document,
-            selection,
-            RangeExpansionBehavior.closedClosed,
-          ),
-        );
+        const boundariesStartSelections = boundaries.map(({ start }) => start);
+        const boundariesEndSelections = boundaries.map(({ end }) => end);
 
-        const sourceMarkSelectionInfos = targets.map((target) =>
-          getSelectionInfo(
-            document,
-            target.contentSelection,
-            RangeExpansionBehavior.closedClosed,
-          ),
-        );
+        const editableEditor = ide().getEditableTextEditor(editor);
 
-        const thatMarkSelectionInfos = targets.map((target) =>
-          getSelectionInfo(
-            document,
-            target.contentSelection,
-            RangeExpansionBehavior.openOpen,
-          ),
+        const contentSelections = targets.map(
+          (target) => target.contentSelection,
         );
 
-        const editableEditor = ide().getEditableTextEditor(editor);
-
-        const [
-          delimiterSelections,
-          cursorSelections,
-          sourceMarkSelections,
-          thatMarkSelections,
-        ] = await performEditsAndUpdateFullSelectionInfos(
-          this.rangeUpdater,
-          editableEditor,
+        const {
+          boundariesStartSelections: delimiterStartSelections,
+          boundariesEndSelections: delimiterEndSelections,
+          sourceSelections: sourceMarkSelections,
+          thatSelections: thatMarkSelections,
+        } = await performEditsAndUpdateSelections({
+          rangeUpdater: this.rangeUpdater,
+          editor: editableEditor,
           edits,
-          [
-            delimiterSelectionInfos,
-            cursorSelectionInfos,
-            sourceMarkSelectionInfos,
-            thatMarkSelectionInfos,
-          ],
-        );
+          selections: {
+            boundariesStartSelections: {
+              selections: boundariesStartSelections,
+              behavior: RangeExpansionBehavior.openClosed,
+            },
+            boundariesEndSelections: {
+              selections: boundariesEndSelections,
+              behavior: RangeExpansionBehavior.closedOpen,
+            },
+            sourceSelections: contentSelections,
```

Comment:
I know it saves lines, but I wonder if we want to rely so heavily on default behavior here. In this case I think it would actually be easier to read if it were parallel to the other entries

Reply:
I can do that for for this specific action, but I don't think we should enforce it in the updated interface.

Reply:
yeah I agree the shorthand is useful sometimes